web/src/codesearch/repo_selector.js

```diff
@@ -37,9 +48,16 @@ function updateOptions(newOptions) {
     }
 
     $('#repos').empty();
+    newOptions.sort()
+
     for (var i = 0; i < newOptions.length; i++) {
-        var option = newOptions[i];
-        $('#repos').append($('<option>').attr('value', option).text(option));
+        var path = newOptions[i].split('/');
+        var group = path.slice(0, path.length - 1).join('/') + '/';
+        var groupQuery = '#repos' + (path.length == 1 ? '' : ' optgroup[data-path="' + group + '"]');
```

Comment:
I have some FUD about using a DOM query here; It's probably fine but relying on the DOM and jQuery selectors to do our grouping is going to be way less efficient (and, IMO, less clear) than just grouping *first* and directly appending the elements we want.

Can we instead use [`_.groupBy`](https://underscorejs.org/#groupBy) to first group by the name, and then just loop over those groups and directly append the `<optgroup>` and `<option>` tags we want? I think I'd expect that to be clearer to read and also a bit faster.

Reply:
I think I found a good compromise. I'd tried to use _.groupBy, but in this approach we need to 
* separate group name and repo name
* group then
* convert result object in array using Object.entries to easily iterate it (also we need sort it again by group name, because this method can crush groups order)
Therefore, much more code is generated and the clearness stay under big question. The result of my try is here:
`newOptions.sort()
    var separatedGroupOptions = newOptions
    .map(function (option) {
        var path = option.split('/');
        return {
            group: path.slice(0, path.length - 1).join('/') + '/', 
            repo: path[path.length - 1]
        }
    });
    var groupedOptions = Object.entries(_.groupBy(separatedGroupOptions, 'group'))
    .map(function (group) {
        var groupName = group[0]
        var clearGroupOptions = group[1].map(function (option) { return option.repo })
        return [groupName, clearGroupOptions]
    });

    for (var i = 0; i < groupedOptions.length; i++) {
        var group = groupedOptions[i][0];
        var options = groupedOptions[i][1];
        var groupDOM = group == '/' ? $('#repos') : $('<optgroup>').attr('label', group)
        for (var j = 0; j < options.length; j++) {
            var option = options[j]
            groupDOM.append($('<option>').attr('value', group + option).text(option))
        }
        if (group != '/') $('#repos').append(groupDOM)
    }
`

Commited way to do it avoids excess dom quering, but save initial locality and readability.